### ISSUE TITLE
Update on how to count

### ISSUE THREAD

**Issue by andkov** [dss-hmi/drug-recedivism-2019#2]
# TL;DR
New report ([./analysis/1-aggregator/1-aggregator][1]) is ready that demonstrate the logic of how we will count convictions. [EDA-1][1] report has been updated to reflect this evolution of thinking. 

# Bullets

1. I thought more about how to count convictions and offenses as I was exploring the data for EDA-2. Before displaying the data, I realized, we need to count the convictions at various aggregation levels. (e.g. person, person-by-date, person-by-date-by-conviction). As the result, I have segregated the process of aggregation (counting the convictions at various level of grouping) into a single script - [`./manipulation/1-aggregator.R`][1].  I have found three person that were useful to demonstrate the logic of aggregation (how we count) and focused on them to display the logic of aggregation in the new report ([./analysis/1-aggregator/1-aggregator.html][1]). 

[1]:https://raw.githack.com/dss-hmi/drug-recedivism-2019/08782371e90779e3af7e5ac3af8e4f76bae4b2c3/analysis/1-aggregator/1-aggregator.html

2.	A better understanding how to count lead me to correct EDA-1, which applied an incorrect logic in counting convictions:
```r
dplyr::mutate(
    n_convictions = length(unique(begin_date)) # convictions, unique date
    ,n_offenses   = n()                        # offenses, counts of
  ) %>% 
```
became  
```r
dplyr::mutate(
    n_convictions       = dplyr::n()         # convictions
    ,n_offense_counts   = sum(offense_count)                        # offenses, counts of
  ) %>% 
```
See more at https://github.com/dss-hmi/drug-recedivism-2019/commit/e083bede42dbee2378dbb45a8449aa77e032aead#diff-9d7d2f67aa6af0dc4bd39799d3c093adL107

# TO DO
@cking34 , please study the [./analysis/1-aggregator/1-aggregator][1] report and make sure that you agree with the operationalization of conviction count ( compare the aggregation tables) 

P.S. I remember about creating a dummy for inmates every having a drug conviction. I didn't include it here, because this value is derivative and will be computed immediately before analysis.
